toddredmond/homework.md

```diff
@@ -0,0 +1,161 @@
+
+**f)** `x = true` - Outputs to true because x was assigned the value of "true".
+
+**g)** `var x; x == 'undefined'` - Outputs to false because the second output cannot be a string in order for this to output to a value of "true"
+
+**h)** `'9'<'10'` - Outputs to false because a string '9' is not less than a string '10'
+
+**i)** `typeof x + 1 === "number"` - Outputs to false because '1' is concatenated to 'x', which would output 'number1', which is not the same as 'number'
+
+**j)** `typeof x % 2 === "number"` - Outputs false because typeof x%2 will output "NaN"
+
+**k)** `typeof (x % 2) === "number"` - Outputs true because the paranthesis cause x%2 to be evaluated first before "typeof" is evaluated.
+
+**l)** `x++ == ++x` - Outputs to false because with x++ the value is incremented, but the existing value is displayed, whereas with ++X, the value is changed and the new value is displayed. 
+
+**m)** `++x == x++` - Outputs true because ++X will increment and output new value, but even though x++ will also increment, it will keep the existing value of x, which will be the same as ++X.
+
+**n)** `"1"+x == 1+x` - Will output to false because the first input will concatenate the values, whereas 1+x will add '1' to the existing value of 'x'.
+
+**o)** `"0"+1 == 1` - Will output to true because '0'+1 and 1 will be close enough, even though the first input is a string and the second is a number. If '===' was used instead, this would evaluate to 'false'
+
+**p)** `(typeof (x+1))===(typeof x)` - Evaluates to true because the parenthesis force (x+1) to process before "typeof" is applied.	
+
+**q)** `(x*1 == x) || ((typeof x) != "number")` - True because (x*1 == x) is true and so whether the next part is true or not, the whole expression is true because || operator was used, which basically equates to "or"
+
+**r)** `(x=(typeof (x+(typeof x))))==x` - True
+
+---
+
+All of the following can be solved with ordinary expressions and global variables with primitive values.  You don't need functions, loops, or other topics beyond our first two classes.
+
+ **2)** (_Difficulty: easy_)
+
+Assume variables x, y, and z are numbers.
+
+**a)**
+Write an expression for the mean (i.e. average) of x, y, and z.
+var x, y, z, mean;
+x=1;
+y=2;
+z=3;
+mean = (x+y+z) / 3;
+
+**b)**
+Write a series of expressions to adjust each of x, y, and z halfway toward their mean.
+That is, reset the value of each variable to something based on its previous value.
+mean = (x+y+z) / 3
+x = x+mea n;
+y = y+mean;
+z = z+mean;
+---
+
+**3)** (_Difficulty: easyish_)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+- _l_ : horizontal position of left edge;
+- _r_ : horizontal position of right edge;
+- _t_ : vertical position of top edge;
+- _b_ : vertical position of bottom edge.
+
+**a)**
+Write an expression for the rectangle's area.
+area = (l+r)*(t+b)
+
```

Comment:
3) Take care to ensure that your expression will work regardless of the position on the graph of your rectangle--for example, if the top left corner is at (5, 10) and the rectangle is a 4 x 3. For example, the length of the top side will be equal to the distance (aka difference) between the left and the right.